src/providers/Products/Products.provider.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { createContext, useContext, useReducer } from 'react';
+
+import productsReducer, { PRODUCTS_ACTIONS } from './productsReducer';
+import { ProductsState } from '../../utils/types';
+
+interface IProductsContext {
+  activeCategories: string[];
+  addCategory: (category: string) => void;
+  removeCategory: (category: string) => void;
+}
+
+interface ProductsProviderProps {
+  children: React.ReactNode;
+}
+
+const initialState: ProductsState = {
+  activeCategories: [],
+};
+
+const ProductsContext = createContext<IProductsContext>({
+  activeCategories: [],
+  addCategory: (category: string) => [],
+  removeCategory: (category: string) => [],
+});
+
+function useProducts() {
+  const context = useContext(ProductsContext);
+  if (!context) {
+    throw new Error(`Can't use "useProducts" without a "ProductsProvider"`);
+  }
+  return context;
+}
+
+function ProductsProvider({ children }: ProductsProviderProps) {
+  const [state, dispatch] = useReducer(productsReducer, initialState);
+
+  const addCategory = (category: string) => {
```

Comment:
Do you think that if we add useCallback to these callback could help us avoid rerenders?